src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs

```diff
@@ -280,6 +283,24 @@ public bool HasPrivateKey
             }
         }
 
+        /// <summary>
+        /// Gets a bool that determines if the 'key_ops' (Key Operations) property should be serialized.
+        /// </summary>
+        /// <return>true if 'key_ops' (Key Operations) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeKeyOps()
+        {
+            return KeyOps.Any();
+        }
+
+        /// <summary>
+        /// Gets a bool that determines if the 'x5c' collection (X.509 Certificate Chain) property should be serialized.
+        ///</summary>
+        /// <return>true if 'x5c' collection (X.509 Certificate Chain) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeX5c()
+        {
+            return X5c.Any();
```

Comment:
Don't need Linq for this. X5c is an IList. If method is needed, just use X5c.Count > 0

Reply:
Just made this change for both methods.